src/app/portlets/content-types/form/content-types-form.component.ts

```diff
@@ -1,160 +1,93 @@
-import {
-    Component,
-    Input,
-    Output,
-    EventEmitter,
-    OnInit,
-    OnChanges,
-    ElementRef,
-    ViewChild,
-    AfterViewInit,
-    OnDestroy
-} from '@angular/core';
+import { Component, ElementRef, EventEmitter, Input, OnInit, Output, ViewChild } from '@angular/core';
 import { FormGroup, FormBuilder, Validators, FormControl } from '@angular/forms';
 
+import { Observable } from 'rxjs/Observable';
+
 import * as _ from 'lodash';
 import { DotcmsConfig } from 'dotcms-js/dotcms-js';
 import { SelectItem } from 'primeng/primeng';
 
-import { BaseComponent } from '../../../view/components/_common/_base/base-component';
+import { ContentTypesInfoService } from '../../../api/services/content-types-info';
 import { DotMessageService } from '../../../api/services/dot-messages-service';
 import { SiteSelectorComponent } from '../../../view/components/_common/site-selector/site-selector.component';
-import { ContentTypesInfoService } from '../../../api/services/content-types-info';
+import { Workflow } from '../../../shared/models/workflow/workflow.model';
+import { WorkflowService } from '../../../api/services/workflow/workflow.service';
 
 // TODO: move this to models
-import { Field } from '../fields';
-import { WorkflowService } from '../../../api/services/workflow/workflow.service';
-import { Workflow } from '../../../shared/models/workflow/workflow.model';
-import { Observable } from 'rxjs/Observable';
-import { DotRouterService } from '../../../api/services/dot-router-service';
+import { ContentTypeField } from '../fields';
 
 /**
-  * Form component to create or edit content types
-  *
-  * @export
-  * @class ContentTypesFormComponent
-  * @extends {BaseComponent}
-  * @implements {OnInit}
-  * @implements {OnChanges}
-  * @implements {AfterViewInit}
-  * @implements {OnDestroy}
-  */
+ * Form component to create or edit content types
+ *
+ * @export
+ * @class ContentTypesFormComponent
+ * @implements {OnInit}
+ */
 @Component({
     providers: [SiteSelectorComponent],
     selector: 'dot-content-types-form',
     styleUrls: ['./content-types-form.component.scss'],
     templateUrl: 'content-types-form.component.html'
 })
-export class ContentTypesFormComponent extends BaseComponent implements OnInit, OnChanges, AfterViewInit, OnDestroy {
```

Comment:
I was able to get rid about all this, because the form now the form shows and hide in a dialog and I put an ```*ngIf``` in the dialog, which means that when the dialog hides the form is destroyed so I don't need to watch changes, everytime the dialog shows the form start on ```ngOnInit```.